shaping/wrapping.go

```diff
@@ -557,36 +557,77 @@ const (
 	newCandidate
 )
 
-// fillUntil tries to fill the provided line candidate slice with runs until it reaches a run containing the
-// provided break option. It returns the index of the run containing the option, the new width of the candidate
-// line, the contents of the new candidate line, and a result indicating how to proceed.
-func (l *LineWrapper) fillUntil(runs RunIterator, option breakOption, startWidth fixed.Int26_6, lineCandidate []Output) (newWidth fixed.Int26_6, newLineCandidate []Output, status fillResult) {
+// lineConfig tracks settings for line wrapping a single line of text.
+type lineConfig struct {
+	// truncating indicates whether this line is being truncated (if sufficiently long).
+	truncating bool
+	// maxWidth is the maximum space a line can occupy.
+	maxWidth int
+	// truncatedMaxWidth holds the maximum width of the line available for text if the truncator
+	// is occupying part of the line.
+	truncatedMaxWidth int
+}
+
+// lineBuilder manages line wrapping candidates.
+type lineBuilder struct {
+	// best tracks the best known line built by this builder within the maxWidth it was used with.
+	best []Output
+	// candidate holds intermediate line state as new lines are built and tested to fit.
+	candidate []Output
+	// candidateWidth tracks the total advance of the runs in candidate.
+	candidateWidth fixed.Int26_6
```

Comment:
Maybe add something like 
it is the cached value of sum(output.Advance for output in candidate)